src/org/jetbrains/plugins/scala/lang/completion/ScalaCaseClassParametersNameContributer.scala

```diff
@@ -0,0 +1,102 @@
+package org.jetbrains.plugins.scala.lang.completion
+
+import com.intellij.codeInsight.completion._
+import com.intellij.codeInsight.lookup.{LookupElementWeigher, LookupElement}
+import com.intellij.patterns.PlatformPatterns
+import com.intellij.psi.PsiElement
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.util.ProcessingContext
+import org.jetbrains.plugins.scala.lang.completion.lookups.ScalaLookupItem
+import org.jetbrains.plugins.scala.lang.psi.api.base.patterns.{ScPattern, ScPatternArgumentList, ScConstructorPattern, ScCaseClause}
+import org.jetbrains.plugins.scala.lang.psi.api.statements.ScFunctionDefinition
+import org.jetbrains.plugins.scala.lang.psi.api.statements.params.ScParameter
+import org.jetbrains.plugins.scala.lang.refactoring.namesSuggester.NameSuggester
+
+/**
+  * Created by kate
+  * on 1/29/16
+  */
+class ScalaCaseClassParametersNameContributer extends ScalaCompletionContributor {
+  extend(CompletionType.BASIC, PlatformPatterns.psiElement(), new CompletionProvider[CompletionParameters] {
+
+    def addCompletions(parameters: CompletionParameters, context: ProcessingContext, _result: CompletionResultSet) {
+      val position = positionFromParameters(parameters)
+      val scope = PsiTreeUtil.getContextOfType(position, classOf[ScCaseClause])
+      if (scope == null) return
+
+      val constructorPattern = PsiTreeUtil.getContextOfType(position, classOf[ScConstructorPattern])
+      if (constructorPattern == null) return
+
+      val classRef = constructorPattern.asInstanceOf[ScConstructorPattern].ref
+      val caseClassParams = classRef.resolve() match {
+        case funcDef: ScFunctionDefinition if funcDef.syntheticCaseClass.isDefined =>
+          funcDef.syntheticCaseClass.get.parameters
+        case _ => return
+      }
+
+      if (caseClassParams.isEmpty) return
+
+      val (corespondedParameter, myPosition) = getCorrespondedParameterForPosition(position, caseClassParams)
+      val result = addByOrderSorter(parameters, _result, myPosition, caseClassParams)
+
+      byClassParamCompletionsItems(caseClassParams, result)
+      byTypeCompletionsItems(position, corespondedParameter, result)
+
+      result.stopHere()
```

Comment:
In this completion  we don't want data from other contributors(like kewordContributer...)